### PR TITLE
Softseg multiclass

### PR DESCRIPTION
**Pull request by andreanne-lemay**
Adapt softseg to multiclass.

To do so, NormReLU is modified when there is more than one class. 
1) ReLU is applied to the model's prediction to get rid of negative values
2) The prediction is divided by the sum in the class dimension. For example, if a given voxel outputs 10 for background, 2 for class 1 and 5 for class 2, the values after activation will because 10/(10 + 2 + 5)=0.588 (background), 2/17=0.118 (class 1) and 5/17 =0.294 (class 2). With softmax, the values would be 0.993 (background), 0.0003 (class 1), 0.0067 (class 2).

This approach is softer and the sum of all classes is always 1. 

Preliminary results on spinal cord and CSF segmentation:
Red: CSF, Blue: SC
The prediction is displayed as an RGB image
image
image

### CODE REVIEW

File: ivadomed/losses.py [hunk 1]
```diff
@@ -377,6 +377,9 @@ def forward(self, input, target):
 
         mask = torch.zeros_like(target)
         kernel = scipy.ndimage.morphology.generate_binary_structure(2, 2)
+        if len(input.shape) == 5:
```
Comment (charleygros) [inline ivadomed/losses.py:380]:
```suggestion
        # For multi-class segmentation tasks
        if len(input.shape) == 5:
```

**Review by charleygros — APPROVED** [review]
Simple and super efficient: Love it!

File: ivadomed/losses.py [hunk 2]
```diff
@@ -377,6 +377,10 @@ def forward(self, input, target):
 
         mask = torch.zeros_like(target)
         kernel = scipy.ndimage.morphology.generate_binary_structure(2, 2)
+        # For multi-class segmentation tasks
+        if len(input.shape) == 5:
```
[the remark this answers is not in the document]
Comment (andreanne-lemay) [reply ivadomed/losses.py:381]: Yes indeed, but this line is required for 3D, my bad! I'll fix the comment

[the remark this answers is not in the document]
Comment (andreanne-lemay) [reply ivadomed/losses.py:381]: Updated in #531
